user-pii-data-updater/src/main/scala/org/sunbird/job/user/pii/updater/function/OwnershipTransferFunction.scala

```diff
@@ -0,0 +1,123 @@
+package org.sunbird.job.user.pii.updater.function
+
+import akka.dispatch.ExecutionContexts
+import org.apache.commons.lang3.StringUtils
+import org.apache.flink.api.common.typeinfo.TypeInformation
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.streaming.api.functions.ProcessFunction
+import org.slf4j.LoggerFactory
+import org.sunbird.job.domain.`object`.{DefinitionCache, ObjectDefinition}
+import org.sunbird.job.user.pii.updater.domain.{ObjectData, OwnershipTransferEvent}
+import org.sunbird.job.user.pii.updater.helpers.{Neo4jDataProcessor, UserPiiUpdater}
+import org.sunbird.job.user.pii.updater.task.UserPiiUpdaterConfig
+import org.sunbird.job.util.{HttpUtil, Neo4JUtil}
+import org.sunbird.job.{BaseProcessFunction, Metrics}
+
+import java.util
+import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+
+class OwnershipTransferFunction(config: UserPiiUpdaterConfig, httpUtil: HttpUtil,
+                                @transient var neo4JUtil: Neo4JUtil = null,
+                                @transient var definitionCache: DefinitionCache = null)
+                               (implicit val stringTypeInfo: TypeInformation[String])
+  extends BaseProcessFunction[OwnershipTransferEvent, String](config) with Neo4jDataProcessor with UserPiiUpdater {
+
+  private[this] val logger = LoggerFactory.getLogger(classOf[OwnershipTransferFunction])
+
+  @transient var ec: ExecutionContext = _
+
+  override def open(parameters: Configuration): Unit = {
+    super.open(parameters)
+    definitionCache = new DefinitionCache()
+    neo4JUtil = new Neo4JUtil(config.graphRoutePath, config.graphName, config)
+    ec = ExecutionContexts.global
+  }
+
+  override def close(): Unit = {
+    super.close()
+  }
+
+  override def metricsList(): List[String] = {
+    List(config.ownershipTransferSuccessEventCount, config.ownershipTransferFailedEventCount, config.ownershipTransferSkippedEventCount)
+  }
+
+  override def processElement(event: OwnershipTransferEvent, context: ProcessFunction[OwnershipTransferEvent, String]#Context, metrics: Metrics): Unit = {
```

Comment:
This method is too big to understand. Can we refactor to small methods?

Reply:
as discussed, will consider refactoring in next release